This is the analysis behind San Francisco Bike Share Program.

In [3]:
%matplotlib inline
import sqlite3
from pandasql import sqldf
import pandas as pd
import matplotlib.pyplot as plt

# df_trip_data_part1 = pd.read_sql_query("""
#                           select * from fact_trips 
#                           left join dim_stations on
#                           fact_trips.start_station_id = dim_stations.id
#                           left join (select * from dim_stations) as dim_stations_end on
#                           fact_trips.end_station_id = dim_stations_end.id
#                           left join (select id, entity_type as Subscriber_Type from dim_entity) as subscriber on
#                           fact_trips.entity_id = subscriber.id
#                           left join (select entity_id, zip_code from dim_entity_zip) as entity_zip_a on
#                           fact_trips.entity_id = entity_zip_a.entity_id
#                           ;
#                           """, 
#                           conn)

df = pd.read_csv("201408_trip_data.csv")
df['Start Date'] = pd.to_datetime(df['Start Date'])
df['End Date'] = pd.to_datetime(df['End Date'])
df['Duration'] = df['End Date'] - df['Start Date']
df['Start Day of Week'] = df['Start Date'].dt.strftime('%A')
df.head()
df.apply(pd.Series.nunique)

Trip ID            171792
Duration             8329
Start Date          93171
Start Station          70
Start Terminal         70
End Date            92115
End Station            70
End Terminal           70
Bike #                681
Subscriber Type         2
Zip Code             3526
dtype: int64

In [5]:
print df['Duration'].mean()

1132.51424979


We have a total of 70 unique terminals in this program.
We have 171792 trips made over a span of 5 months, from March 2014 to August 2014.
The average trip takes about 18 minutes


Busiest Stations:

Time/Day Analysis: 

Weekdays vs Weekends

Implication? Should we increase or decrease capacity on weekdays or weekends

Knowing our users

Percentage share of customers vs subscribers

DO they have different behaviour, usage times.

Which zip codes do they come from

Which terminals do they come from